package/src/frontend/media/media.js

```diff
@@ -33,3 +44,4 @@ export const media = {
 };
 
 Object.assign(media, BackboneEvents);
+document.addEventListener('visibilitychange', media.muteInBackground, false);
```

Comment:
This will break server side rendering and also run in Paged where it is not needed. Maybe we can set this up when media is first unmuted? Since `muteInBackground` is never supposed to be called by outside code, I'd test it by dispatching `visibilitychange` events on document.

Reply:
Do you know how can I dispatch event to hide page?
i have tried this 
`  window.dispatchEvent(new Event('pagehide'));`
but it does not work

Reply:
You listen on `document`, so I'd guess you need to use `document.dispatchEvent`. Also the event name is different.

Reply:
I had tried it with document also. what would be the event name?
tried this also `document.dispatchEvent(new Event('visibilitychange'));`

Reply:
Initially I tried testing through document visibility change but I could not find any solution for that.
after that I decided to test it manually

Reply:
This passes on my machine:
```js
const listener = jest.fn();

document.addEventListener('visibilitychange', listener);
document.dispatchEvent(new Event('visibilitychange'));
      
expect(listener).toHaveBeenCalled();
```

Reply:
Listener gets called for me as well. I am testing muteInBackground function and visibilityState property of document does not change through this event.

Reply:
see https://github.com/jsdom/jsdom/issues/2190

Reply:
defineProperty is already being used. I will use both approaches to makes the tests complete